source/pdo_sqlsrv/core_stmt.cpp

```diff
@@ -1960,7 +1960,7 @@ void default_sql_size_and_scale( sqlsrv_stmt* stmt, unsigned int paramno, zval*
                 column_size = SQL_SERVER_MAX_TYPE_SIZE;
             }
             else {
-				column_size = SQL_SERVER_MAX_FIELD_SIZE / char_size;
+				column_size = (char_size == 0) ? 1 : (SQL_SERVER_MAX_FIELD_SIZE / char_size);
```

Comment:
`char_size` is computed differently on Windows:
* [PHP-7.0/sqlsrv/core_stmt.cpp line 1924](https://github.com/Microsoft/msphpsql/blob/PHP-7.0/sqlsrv/core_stmt.cpp#L1924)
* [PHP-7.0-Linux/source/sqlsrv/core_stmt.cpp line 1957)](https://github.com/Microsoft/msphpsql/blob/PHP-7.0-Linux/source/sqlsrv/core_stmt.cpp#L1957)

Maybe not multiplying with the parameter string size would be a better fix?
Explaining the reason why the code changed in 4.0.3 would also help.